lib/services/dialog/index.ts

```diff
@@ -155,9 +158,15 @@ class DialogManagement extends AbstractManager<{ utils: typeof utils }> implemen
 
       try {
         const prefix = dmPrefix(dmStateStore.intentRequest.payload.intent.name);
-        const { intentClassificationSettings, intents, isTrained, slots } = castToDTO(version, project);
+        const { intents, isTrained, slots } = castToDTO(version, project);
+
+        const scopedIntent = intents?.find((intent) => intent.name === dmStateStore.intentRequest!.payload.intent.name);
+        const scopedSlots = scopedIntent?.slots?.map((slot) => slots?.find((entity) => entity.key === slot.id));
```

Comment:
```suggestion
        const scopedSlots = scopedIntent?.slots?.map((slot) => slots?.find((entity) => entity.key === slot.id)).filter(Boolean);
```
then below you don't have to `slot?.name`

Reply:
Typescript isn't smart enough to know that slot is not undefined here